### PR TITLE
Fixed migration failing when specific duplicate prefixes are present.

### LINKED ISSUE #5334: Migration ipam.0022 fails when specific duplicate IPAM objects are present
### Environment
* Nautobot version (Docker tag too if applicable): v2.1.6b1
* Python version: 
* Database platform, version: PG16
* Middleware(s):

### Steps to Reproduce
1. In 1.6, using the same prefix and prefix length, create an aggregate, a prefix in the Global VRF and a prefix in another VRF
2. Upgrade to any version of 2.x and run `nautobot-server migrate`

### Expected Behavior
Migrations succeed

### Observed Behavior
Migrations fail on ipam.0022

```
  Applying ipam.0022_aggregate_to_prefix_data_migration...Traceback (most recent call last):
  File "/usr/local/bin/nautobot-server", line 6, in <module>
    sys.exit(main())
  File "/source/nautobot/core/cli/__init__.py", line 52, in main
    run_app(
  File "/source/nautobot/core/runner/runner.py", line 297, in run_app
    management.execute_from_command_line([runner_name, command, *command_args])
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.10/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/source/nautobot/ipam/migrations/0022_aggregate_to_prefix_data_migration.py", line 161, in migrate_aggregate_to_prefix
    prefix = Prefix.objects.get(network=instance.network, prefix_length=instance.prefix_length)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 439, in get
    raise self.model.MultipleObjectsReturned(
__fake__.Prefix.MultipleObjectsReturned: get() returned more than one Prefix -- it returned 2!
```

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes #5334
# What's Changed

# Screenshots

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/5334.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed migration failing when specific duplicate prefixes are present.
```
Comment (glennmatthews) [inline changes/5334.fixed:1]:
```suggestion
Fixed migration from 1.x failing when specific duplicate prefixes are present.
```

File: nautobot/ipam/migrations/0022_aggregate_to_prefix_data_migration.py [hunk 2]
```diff
@@ -158,7 +158,11 @@ def migrate_aggregate_to_prefix(apps, schema_editor):
             prefix_length=instance.prefix_length,
             vrf__isnull=True,
         ).exists():
-            prefix = Prefix.objects.get(network=instance.network, prefix_length=instance.prefix_length)
+            prefix = Prefix.objects.filter(
+                network=instance.network,
+                prefix_length=instance.prefix_length,
+                vrf__isnull=True,
+            ).first()
```
Comment (glennmatthews) [inline nautobot/ipam/migrations/0022_aggregate_to_prefix_data_migration.py:165]: Is `first` correct/sufficient here or do we need to handle *all* matching Prefixes?

Comment (gsnider2195) [reply nautobot/ipam/migrations/0022_aggregate_to_prefix_data_migration.py:165]: This is trying to find a suitable prefix to migrate the aggregate fields to (date_allocated and rir). It's unlikely that someone has an aggregate with the same network/length as duplicate prefixes without a vrf. We'd probably want to tie in to the namespace migrations if we were going to try to account for that situation. The more likely scenario (which was encountered and prompted the bug report) is where they have an aggregate, a prefix without a vrf and a prefix with a vrf that all have the same network/length.

Comment (glennmatthews) [reply nautobot/ipam/migrations/0022_aggregate_to_prefix_data_migration.py:165]: Unlikely sure but without `ENFORCE_GLOBAL_UNIQUE=True` it's certainly possible in 1.x. Probably not worth losing sleep over though. Thanks.

**Review by HanlinMiao — APPROVED** [review]

**Pull request merged**